hibike/spawn_virtual_devices.py

```diff
@@ -0,0 +1,37 @@
+import csv
+import subprocess
+import shlex
+import re
+import os
+import atexit
+import time
+
+popens = []
+
+def get_virtual_ports():
+    socat = subprocess.Popen(shlex.split("socat -d -d pty,raw,echo=0 pty,raw,echo=0"), stderr=subprocess.PIPE)
+    popens.append(socat)
+    lines = [str(socat.stderr.readline()), str(socat.stderr.readline())]
```

Comment:
I think readline() was returning bytestrings but I wanted unicode for some reason